framework/Optimizers/gradients/FiniteDifference.py

```diff
@@ -56,7 +56,7 @@ def getInputSpecification(cls):
   ###############
   # Run Methods #
   ###############
-  def chooseEvaluationPoints(self, opt, stepSize):
+  def chooseEvaluationPoints(self, opt, stepSize, constraints=None):
```

Comment:
should it be constraints?
If so, can you update the doctring and declaration to all the other methods? 

In addition, shouldn't we perform the same check in the other "gradients" classes? (maybe with a similar strategy?)?

Reply:
Unfortunately the strategy is not as trivial with the other strategies; while it would not be too hard to come up with a mathematically sound solution, it is more time than I currently have to devote to it right now. I hope I can return to this soon with #1412.